### DISCUSSION TITLE
IDE suggestion

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#11271] [Using MicroPython]
Greeting micropythonistas! 

I started my upython journey back in 2016 so Atom with pymakr is my default IDE. Recently I wanted to try out a wemos S2 mini. For some reason Atom won't connect with the mini (could it be something to do with the mini using uart0 for usb?) so I swapped to Thonny which crashes with
```
>>> THONNY FAILED TO EXECUTE COMMAND get_globals


SCRIPT:
__thonny_helper.print_mgmt_value({name : (__thonny_helper.repr(value), id(value)) for (name, value) in globals().items() if not name.startswith('__')})

STDOUT:


STDERR:
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <dictcomp>
TypeError: 'str' object isn't callable
```
for a simple program that ends
```
for i in range (9):
 if uart.any(): reply=uart.read(160); print('reply', reply); break
```
So now I'm in need of an IDE. VScode looks way too complicated so I'm after something simple & reliable that works with an S2 mini & doesn't have a hangup with print statements? Any suggestions?

**Comment by mattytrentini** [comment]
My preference is VS Code, using [mpremote](https://docs.micropython.org/en/latest/reference/mpremote.html) in a terminal to synchronise code to a device. 

If VS Code is too complex, choose your own IDE (Atom is fine, though development has been discontinued) but, again, use mpremote to interact with your device.

**Reply by fabiomolinar** [reply]
Hi @mattytrentini, I am also using VS Code. However, I didn't figure out a way to set `micropython` as the python interpreter so I can avoid getting warning messages that the modules I am using are not found. I know I can set up the interpreter in VS Code, but it usually expects a `.exe` file, and on `micropython` webpage I can only get the `.bin` files. So, what do you do in this case?

**Reply by Josverl** [reply]
To have VSCode (and most other mature IDEs that can work with an LSP) 'understand' MicroPython modules, extensions and limitations   
you probably want to install the [micropython-stubs](https://github.com/Josverl/micropython-stubs?tab=readme-ov-file#micropython-stubs-enhance-your-development-experience)

**Reply by fabiomolinar** [reply]
Thanks for the reference!

**Comment by ma261065** [comment]
Have you tried [Viper IDE](http://viper-ide.org) ?
